Source/WebCore/Modules/mediasession/MediaSession.cpp

```diff
@@ -239,11 +239,13 @@ void MediaSession::setActionHandler(MediaSessionAction action, RefPtr<MediaSessi
 {
     if (handler) {
         ALWAYS_LOG(LOGIDENTIFIER, "adding ", action);
+        Locker lock { m_actionHandlersLock };
```

Comment:
Still too wide a scope. We should only include the HashMap operation, not unrelated operation and external function calls.